### PR TITLE
Cypress setup fix

### PR DESCRIPTION
**Pull request by chaim-aharonson-applitools**
> Following [this PR](https://github.com/applitools/eyes.sdk.javascript1/pull/1020) of supporting a 'caret' in Cypress version

* 'Cypress' version in 'npx eyes-setup' script should be verified against the actual installed 'Cypress' version, aka: `node_modules/cypress/package.json -> version` (and not as it was: against the 'project' `package.json -> version`)
* The **test** for 'npx eyes-setup' script was moved from 'it' to 'e2e' since 'Cypress' installation is required.
* The logic of parsing and validating the version by using `semver` in `/bin/eyes-setup.js` remained the same as before (although in current implementation, version should never have a caret) .
* No change-log since the initial fix functionality ([this PR](https://github.com/applitools/eyes.sdk.javascript1/pull/1020)) was **not** released yet and this branch only improves that functionality, original change-log record is: _Fix `npx eyes-setup` for Cypress version with a caret, support `e2e.ts` as support file_
* Tests are failing, this is being handled in a different Branch

### CODE REVIEW

**Review by SophieDePaula — APPROVED** [review]
looks good

**Review by amitzur — APPROVED** [review]
Overall this is great!
I had a few comments, but generally this is approved.

BTW, in the first run it took this test 4 minutes, because it was installing Cypress from scratch (both versions 9 and 10). In the second run it took 13 seconds...
I wonder how much it would take in the CI.

File: js/packages/eyes-cypress/bin/eyes-setup.js [hunk 1]
```diff
@@ -6,23 +6,19 @@ const {handlePlugin} = require('../src/setup/handlePlugin');
 const {handleCommands, handlerCommandsCypress10} = require('../src/setup/handleCommands');
 const {handleTypeScript, handlerTypeScriptCypress10} = require('../src/setup/handleTypeScript');
 const {version} = require('../package');
-const fs = require('fs');
+const {resolve, dirname} = require('path');
 const cwd = process.cwd();
+const fs = require('fs');
+const cypressPackage = require.resolve(cwd + '/node_modules/cypress');
```
[the remark this answers is not in the document]
Comment (chaim-aharonson-applitools) [reply js/packages/eyes-cypress/bin/eyes-setup.js:12]: this implementation uses the `packages/eyes-cypress` installation and not the REAL one, for example: running `npx eyes-setup` with the fix you suggested from a tutorial will not use the project cypress installed version but the `eyes` version

[the remark this answers is not in the document]
Comment (amitzur) [reply js/packages/eyes-cypress/bin/eyes-setup.js:12]: `@applitools/eyes-cypress` should not have `cypress` in its installed `node_modules` folder. I am not sure what you say here is true.

[the remark this answers is not in the document]
Comment (chaim-aharonson-applitools) [reply js/packages/eyes-cypress/bin/eyes-setup.js:12]: > `@applitools/eyes-cypress` should not have `cypress` in its installed `node_modules` folder. I am not sure what you say here is true.

`@applitools/eyes-cypress` has `devDependencies` for 'cypress', so local installation on dev machine has `cypress` under 'node_modules'.
This issue was re-verified, once running `npx eyes-setup` for a tutorial (as an example), and using the suggested 
```
const cypressPackage = require.resolve('cypress');
```
setup identify the wrong version (in this case version 8 as in `eyes-cypress -> devDependencies -> cypress`), replacing to original commit fixes that and correct version is detected.
It also caused a failure in e2e test `test/e2e/setup/eyes-setup.e2e.test.js`.
Additionally, once trying to reproduce a bug, it caused a test of 'cypress version >= 10' to pass in spite the content of 'cypress.config.js' file was wrong

[the remark this answers is not in the document]
Comment (chaim-aharonson-applitools) [reply js/packages/eyes-cypress/bin/eyes-setup.js]: @amitzur `require.resolve` does have `options` option. now it works as expected by running both the e2e tests and from cypress tutorial

**Comment by SophieDePaula** [comment]
@amitzur do you have any additional comments on this PR or can I merge it?

File: js/packages/eyes-cypress/src/setup/getCypressPaths.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+function getCypressPaths({cwd, isCypress10}) {
+  if (isCypress10) {
+    return getCypressPaths10AndAbove(cwd);
+  } else {
+    return getCypressPaths9AndBelow(cwd);
+  }
+}
+
+function getCypressPaths10AndAbove(cwd) {
+  const cypressConfigPath = path.resolve(cwd, 'cypress.config.js');
```
Comment (SophieDePaula) [inline js/packages/eyes-cypress/src/setup/getCypressPaths.js:15]: @amitzur shouldn't this be `cypress.config.js` or `cypress.config.ts`?

Comment (SophieDePaula) [reply js/packages/eyes-cypress/src/setup/getCypressPaths.js:15]: Oh nvm, I'll need to add this to my PR.

**Review by SophieDePaula — APPROVED** [review]
looks good!

File: js/packages/eyes-cypress/src/setup/getCypressPaths.js (hunk 2 above)
Comment (amitzur) [reply js/packages/eyes-cypress/src/setup/getCypressPaths.js:15]: @SophieDePaula Exactly. When you add support for `cypress.config.ts` then you will need to look for both :)

**Pull request merged**